apollo-router/src/json_ext.rs

```diff
@@ -21,6 +24,11 @@ pub(crate) type Object = Map<ByteString, Value>;
 
 const FRAGMENT_PREFIX: &str = "... on ";
 
+static TYPE_CONDITIONS_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"(?:\|\[)(.+?)(?:,\s*|)(?:\])")
```

Comment:
NIT: Maybe easier to understand when captures are used if these are named.

Reply:
fixed in d14d077d0

Reply:
I'm not sure this is really fixed. I can see that you've named one of the groups "<condition>", but I can't see where it's being used. It looks like you are still using the indices of the various captures.

e.g.:
```
        let path = TYPE_CONDITIONS_REGEX.replace(s, |caps: &Captures| {
            type_conditions.extend(
                caps.extract::<1>()
                    .1
                    .map(|s| s.split(',').map(|s| s.to_string()))
                    .into_iter()
                    .flatten(),
            );
            ""
        });
```
You are always using `extract, group 1`, which is a bit opaque. If you want to use the name then it would be something like: `caps.name("condition")` and you'd need to do some handling for not being present.

Anyway, it was a NIT, so I'm fine with the `extract` approach that you are using, in which case no need to name the capture group.

If you do think it's worth naming the various capture groups, then you should use them to make the code a bit less opaque when you are pulling out the conditions as I outlined above.

Reply:
actually fixed in 6c67cacfe 😁

Reply:
Now you are making me feel really picky, but there multiple used of `extract()`. May as well change all of them.